pkg/cmd/cluster/connect.go

```diff
@@ -463,3 +486,22 @@ func getOneHeadlessEndpoint(clusterDef *appsv1alpha1.ClusterDefinition, secrets
 	}
 	return string(val)
 }
+
+// getClusterCharacterType will get the cluster character type compatible with 0.7
+// If both componentDefRef and componentDef are provided, the componentDef will take precedence over componentDefRef.
+func (o *ConnectOptions) getClusterCharacterType() error {
+	if o.component.ComponentDef != "" {
+		o.componentDefV2 = &appsv1alpha1.ComponentDefinition{}
+		if err := util.GetK8SClientObject(o.Dynamic, o.componentDefV2, types.CompDefGVR(), "", o.component.ComponentDef); err != nil {
+			return err
+		}
+		o.characterType = o.componentDefV2.Spec.ServiceKind
```

Comment:
The characterType required for the connection here should be a custom defined by kbcli. The original intent of ServiceKind design does not fully align with the meaning of characterType. In the long run, it should not rely on ServiceKind.